### DISCUSSION TITLE
Numeric display mode and Unicode text?

### DISCUSSION THREAD

**Discussion by wingkinl** [jovibor/HexCtrl#71] [Ideas]
![image](https://user-images.githubusercontent.com/26253549/142007994-6407bcab-9b3e-41e8-bbd6-cae6cd325180.png)

Just like the Memory window in Visual Studio that allows us to switch between hex/numeric mode, I wonder if this already exists in the code, or do you have plan on supporting this?

Unicode text mode would be another quite useful feature, since we often have to deal with it on Windows.

Is there any particular difficulty in implementing these features? I may also try and see if I can contribute to this project.

Great project, by the way.

**Comment by jovibor** [comment]
> Just like the Memory window in Visual Studio that allows us to switch between hex/numeric mode, I wonder if this already exists in the code, or do you have plan on supporting this?

This would require redesigning the whole rendering pipeline. While it's generally possible of course, it's not something that's in my nearest TODO list.

> Unicode text mode would be another quite useful feature, since we often have to deal with it on Windows. Is there any particular difficulty in implementing these features? 

Since latest 15edd0d commit it's already supported, in experimental-limited mode. The difficulty in Unicode support is that (especially UTF-8 and all other variadic encodings) you never know how many code-units a next code-point will have. In a text file you just know that this is a text, raw binary data is absolutely different story. You don't know whether it's text or not, and if it's a text, how many bytes it occupies. Basically it all leads to many calculations and checks.

> I may also try and see if I can contribute to this project.

You're always welcome.
